gnucash/gnome/window-reconcile.c

```diff
@@ -851,6 +852,11 @@ startRecnWindow(GtkWidget *parent, Account *account,
         if (gnc_reverse_balance(account))
             *new_ending = gnc_numeric_neg (*new_ending);
 
+        if ((*statement_date > gnc_time64_get_today_end ()) &&
+            (!gnc_verify_dialog (GTK_WINDOW (parent), FALSE,
+                                 _("Statement date is in the future. This is very likely to be an error. Using future statement dates is very likely to cause reconciliation difficulties in the future. Please confirm you wish to use a future statement date."))))
+            goto restart_dialog;
+
```

Comment:
This is the wrong place to check. Do it inside the dialog code and disable OK.

This is a fine preventative measure going forward but doesn' t do anything for users whose books already have bad reconcile dates.

Reply:
> This is the wrong place to check. Do it inside the dialog code and disable OK.
No, disabling OK means that you break a useful feature of being able to reconcile a day or two ahead which I consider important.  I still think allowing 1 week into the future, albeit arbitrary, is perfectly reasonable.  Or perhaps split the difference?  Allow 1 week into the future without a popup, and if it is > 1 week then pop up the warning (with or without "OK").
I can tell you now that I **will** change the code myself to allow future reconciles if they get turned off.

Reply:
I'd still vote for *any* future reconcile date pops a loud warning on OK: "Are you sure?". Only @derekatkins will face needing an extra click... I tried to do it inside the date_changed signal handler but it is either too obnoxious (click after each date change) or not visible enough (a GtkLabel appearing/disappearing).

Reply:
The solution to that is to include the "remember this" and "remember this for this session" buttons. I don't know offhand where they live, but @Bob-IT  probably does.

Reply:
What you want is to create a dialog and use 'gnc_dialog_run' which use gnc-warnings.h/c and also schema org.gnucash.warnings.gschema.xml.in

Do a search for 'gnc_dialog_run' for examples.